zingolib/src/wallet/transaction_record.rs

```diff
@@ -379,6 +347,44 @@ impl TransactionRecord {
             })
         })
     }
+
+    /// get a list of unspent NoteIds with associated note values
+    pub(crate) fn get_spendable_note_ids_and_values(
+        &self,
+        sources: &[zcash_client_backend::ShieldedProtocol],
+        exclude: &[NoteId],
+    ) -> Vec<(NoteId, u64)> {
+        let mut all = vec![];
+        if sources.contains(&Sapling) {
+            self.sapling_notes.iter().for_each(|zingo_sapling_note| {
+                if zingo_sapling_note.is_unspent() {
+                    if let Some(output_index) = zingo_sapling_note.output_index() {
+                        let id = NoteId::new(self.txid, Sapling, *output_index as u16);
+                        if !exclude.contains(&id) {
+                            all.push((id, zingo_sapling_note.value()));
+                        }
+                    } else {
+                        println!("note has no index");
```

Comment:
What case does this match?

Reply:
loaded an old wallet with incomplete information about notes... maybe we could create an error case that automatically rescans this transaction?